AutomatedTesting/Gem/PythonTests/scripting/NodeCategory_ExpandOnClick.py

```diff
@@ -45,79 +55,90 @@ def NodeCategory_ExpandOnClick():
+        click_position = QtCore.QPoint(rect_left_x, rect_center_y)
+        #click position relative to node palette tree view and not screen space xy
+        QtTest.QTest.mouseClick(node_palette_node_tree.viewport(),
+                                QtCore.Qt.LeftButton,
+                                QtCore.Qt.NoModifier,
+                                click_position,
+                                )
+
+    def double_click_node_category(self, node_palette_node_tree, index):
+        item_index_center = node_palette_node_tree.visualRect(index).center()
         # Left click on the item before trying to double click, will otherwise fail to expand
         # as first click would highlight and second click would be a 'single click'
-        pyside_utils.item_view_index_mouse_click(item_view, index)
-        QTest.mouseDClick(item_view.viewport(), QtCore.Qt.LeftButton, QtCore.Qt.NoModifier, item_index_center)
+        pyside_utils.item_view_index_mouse_click(node_palette_node_tree, index)
+        QtTest.QTest.mouseDClick(node_palette_node_tree.viewport(),
+                                 QtCore.Qt.LeftButton,
+                                 QtCore.Qt.NoModifier,
+                                 item_index_center)
+
+    @pyside_utils.wrap_async
+    async def run_test(self):
 
-    # 1) Open Script Canvas pane
-    general.open_pane("Script Canvas")
-    Report.critical_result(Tests.pane_open, general.is_pane_visible("Script Canvas"))
+        # Preconditions
+        general.idle_enable(True)
+        general.close_pane(SCRIPT_CANVAS_UI)
+        helper.wait_for_condition(lambda: general.is_pane_visible(SCRIPT_CANVAS_UI) is None, WAIT_TIME_3)
 
-    # 2) Get the SC window objects
-    editor_window = pyside_utils.get_editor_main_window()
-    sc = editor_window.findChild(QtWidgets.QDockWidget, "Script Canvas")
-    if sc.findChild(QtWidgets.QDockWidget, "NodePalette") is None:
-        action = pyside_utils.find_child_by_pattern(sc, {"text": "Node Palette", "type": QtWidgets.QAction})
-        action.trigger()
-    node_palette = sc.findChild(QtWidgets.QDockWidget, "NodePalette")
-    nodeTree = node_palette.findChild(QtWidgets.QTreeView, "treeView")
-    ai_index = pyside_utils.find_child_by_pattern(nodeTree, "AI")
+        # 1) Open Script Canvas pane
+        general.open_pane(SCRIPT_CANVAS_UI)
+        Report.critical_result(Tests.pane_open, general.is_pane_visible(SCRIPT_CANVAS_UI))
 
-    # 3) Ensure all categories are collapsed for a clean state
-    nodeTree.collapseAll()
+        # 2) Get the SC window objects
+        scripting_tools.initialize_editor_object(self)
+        scripting_tools.initialize_sc_editor_objects(self)
+        if self.sc_editor.findChild(QtWidgets.QDockWidget, NODE_PALETTE_QT) is None:
+            action = pyside_utils.find_child_by_pattern(self.sc_editor, {"text": NODE_PALETTE_UI, "type": QtWidgets.QAction})
+            action.trigger()
```

Comment:
Extract to meaningfully named helper function

Reply:
extracted to scripting tools file.